protocol/protobuf/chat_identity.proto

```diff
@@ -58,6 +58,10 @@ message IdentityImage {
   // encrypted signals the encryption state of the payload, default is false.
   bool encrypted = 5;
 
+  // version iterator of the image since images use a localUrl to be shown,
+  // this iterator makes sure the image URL changes when a new image is given
+  uint32 version = 6;

```

Comment:
Wait, so `version` is not just local, it becomes part of `CommunityDescription`? 🤔 
Sounds a bit weird to be like that, like this problem should be solved solely on client side, not affecting network messages

Reply:
Yeah I agree it should be local only, but I didn't find an easy way to do so, since locally, we use the `IdentityImage` protobuf in the image server.

I'm very open for suggestions to make it locally, but apart from creating some sort of new column or table, I don't know how to make it work.

In the end, this solution is pretty unobtrusive anyway IMO.

Reply:
@jrainville I think I'm not convinced, modifying protocol for local needs sounds counter-intuitive to me.
Ww should probably wrap the `IdentityImage` when passing it to `ServerMedia`, and add the `Version` property to the wrapper. Dunno, or make some method of the media server to update the version of specific image.

> _creating some sort of new column or table_

We don't need to persist the version, it's enough to have it only during a single run. After restart, version can go again from `1`, should work, right?

Reply:
> We don't need to persist the version, it's enough to have it only during a single run. After restart, version can go again from 1, should work, right?

That's true. I'll try to see if I can do as you say

Reply:
I agree with Igor here.

At first glance, the best would be to use/extend [CommunityChanges](https://github.com/status-im/status-go/blob/ad28f15531c76c216f08118d4055a140ee52f80f/protocol/communities/community_changes.go#L20) with image change recognition and then populate it to client.

Reply:
> We don't need to persist the version, it's enough to have it only during a single run. After restart, version can go again from 1, should work, right?

Ok, I tried something simple and I realized that we indeed need to save it somewhere. At least a local cache. However, we do not have a local cache for communities. The closest we have is the `encrypted_community_description_cache` which is a table. It is also what the image server uses to serve the images.

Just to make sure the problem is clear, I can't just pass the updated version to the image server, since the Edit happens in one thread/timeframe and the image serving is a completely different one, so the image server knows nothing about image updates.
It serves using the community cache table, which is populated from the decrypted community description. So we need the version or at least an indicator that the images were updated in there.

So either I create a new local cache for image versions per community, or I add a new column `encrypted_community_description_cache` to save the version, or leave it as I did it. Or you guys have a better option?